.github/actions/promote_docker_image/action.yml

```diff
@@ -0,0 +1,65 @@
+name: Promotes built Docker images to target environment registries for consumption
+
+inputs:
+  project_name:
+    required: true
+  codeowner:
+    required: true
+  image_name:
+    required: true
+  image_tags:
+    required: true
+    description: A comma separated list of image tags to promote.
+  source_account_id:
+    required: false
+    default: "081157560428"
+  source_region:
+    required: false
+    default: us-east-1
+  target_account_id:
+    required: true
+  target_region:
+    required: true
+  target_environment:
+    required: true
+    default: dev
+
+runs:
+  using: "composite"
+  steps:
+    - name: configurate_source_account_aws_credentials
+      uses: aws-actions/configure-aws-credentials@v3
+      with:
+        role-to-assume: arn:aws:iam::${{ inputs.source_account_id }}:role/github-main-role
+        role-session-name: main-role-session-${{ inputs.image_name }}
+        aws-region: us-east-1
+    - name: ecr_login
+      uses: aws-actions/amazon-ecr-login@v2
+    - name: pull_source_images
+      shell: bash
+      run: |
+        docker pull 081157560428.dkr.ecr.us-east-1.amazonaws.com/terraform-deploy-slim:lts-0
```

Comment:
Assuming `081157560428` should be `${{ inputs.source_account_id }}` instead?

Reply:
Well, not necessarily. Terraform deploy will always be hosted in Infra. The source account id in this context refers to the built image the user is deploying.

Reply:
you're using the source_account_id to log into AWS before this, so this will fail if that's not set to infra.